### DISCUSSION TITLE
Import RT-Plan

### DISCUSSION THREAD

**Discussion by NatukiAdachi** [e0404/matRad#754] [Q&A]
Hello. 
I want to import CT, RT-Struct, Dose, Plan, but when I try to import the RT-plan I get the following errorAny solution?Is there any solution?

`フィールド名 "Item_2" が認識されません。

エラー: matRad_importFieldShapes (行 48)
    tmpCollimation.beamMeterset(i) = fractionSequence.ReferencedBeamSequence.(beamSeqNames{i}).BeamMeterset;

エラー: matRad_importDicomRTPlan (行 124)
    pln.propStf.collimation  = matRad_importFieldShapes(BeamSequence,fractionSequence);

エラー: matRad_importDicom (行 106)
        pln = matRad_importDicomRTPlan(ct, files.rtplan, dicomMetaBool);

エラー: matRad_importDicomGUI>import_button_Callback (行 330)
matRad_importDicom(files, dicomMetaBool);

エラー: gui_mainfcn (行 95)
        feval(varargin{:});

エラー: matRad_importDicomGUI (行 53)
    [varargout{1:nargout}] = gui_mainfcn(gui_State, varargin{:});

エラー: matlab.graphics.internal.figfile.FigFile/read>@(hObject,eventdata)matRad_importDicomGUI('import_button_Callback',hObject,eventdata,guidata(hObject))`

**Comment by wahln** [comment]
This is difficult to debug remotely without the dicom file, especially since I don't know how to interpret Japanese error messages.
I would guess that matRad tries to find Beam MU for a beam but for some reason it does not find the DICOM tag BeamMeterset in the BeamSequence. 
Actually, we should not require this setting because [it is not required by the standard](https://dicom.innolitics.com/ciods/rt-plan/rt-fraction-scheme/300a0070/300c0004/300a0086).
What kind of Plan do you try import? IMRT? VMAT? Maybe I can think of a code fix...

**Comment by NatukiAdachi** [comment]
Thanks for the reply.

The treatment plan I tried to import is IMRT.
I received this data from TCIA, but when I try to import data from other
TCIA's it does not import properly. Is there some kind of compatibility?

2024年8月5日(月) 23:26 Niklas Wahl ***@***.***>:

**Comment by wahln** [comment]
If it is from TCIA, I may be able to download the same dataset and try to import it? Which dataset from TCIA is it?

As a quick workaround, you could replace the following lines in matRad_importFieldShapes.m:

```matlab
% get total MU applied by beam i
tmpCollimation.beamMeterset(i) = fractionSequence.ReferencedBeamSequence.(beamSeqNames{i}).BeamMeterset;
```
with an if/else statement to set a default value:
```matlab
% get total MU applied by beam i
if isfield(fractionSequence.ReferencedBeamSequence.(beamSeqNames{i}),'BeamMeterset')
    tmpCollimation.beamMeterset(i) = fractionSequence.ReferencedBeamSequence.(beamSeqNames{i}).BeamMeterset;
else
    tmpCollimation.beamMeterset(i) = 1; %Set Default BeamMeterset;
end
```

This might at least do the import for now, however, at the same time lead to wrong units. Let me know if other issues pop up during the import.

**Reply by wahln** [reply]
@amitantony do you understand what _should_ happen if the `BeamMeterset` is not present in the `ReferencedBeamSequence`?
Should this just be replaced by `FinalCumulativeMetersetWeight` in the `BeamSequence`? [This link here](https://dicom.innolitics.com/ciods/rt-plan/rt-beams/300a00b0/300a010e) gives a rule for calculation, but there's no specification of what should be assumed when `BeamMeterset` is not given...

**Reply by wahln** [reply]
Currently I think that if it is not given, we need to take the value of `FinalCumulativeMetersetWeight` as the `BeamMeterset`. So a sustainable fix would store that the value was not present (for example by storing an empty array), and then in this code block not take the relative value:
https://github.com/e0404/matRad/blob/2b499dfbfbd37660c46afb8acded4353f762c1c3/matRad/dicom/matRad_importFieldShapes.m#L139-L149

**Reply by wahln** [reply]
But maybe we wait with a fix until the refactor is complete.

**Comment by NatukiAdachi** [comment]
2024年8月6日(火) 17:19 Niklas Wahl ***@***.***>:
Dataset : HNC-IMRT-70-33
I will try it

**Comment by NatukiAdachi** [comment]
Thank you for your response.

2024年8月6日(火) 17:40 Niklas Wahl ***@***.***>:
